assets/src/apps/app.tsx

```diff
@@ -1,76 +1,51 @@
 import { ModalDisplay } from 'components/modal/ModalDisplay';
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React, { FC } from 'react';
 import { Provider } from 'react-redux';
-import { State } from 'state/index';
 import { configureStore } from 'state/store';
 import { Maybe } from 'tsmonad';
-import { b64DecodeUnicode } from 'utils/decode';
 
-export function defineApplication<T extends State>(Component: React.FunctionComponent<any>) {
-  // TODO, allow a customized, per app state (both initial state and collection of reducers)
-  // to be passed into this function, instead of simply using a shared common state
-  let store = configureStore();
+/**
+ * Registers a client side react/js application to be used in Phoenix templates via `ReactPhoenix.ClientSide.react_component`
+ * This will wrap the component in a redux store and give it a ModalDisplay top level sibling component it then registers
+ * that as window.Component.*name*
+ *
+ * @param name Name of the app to register. Is used in both the key on window.Components and the debugging label in react-dev-tools
+ * @param Component Any react component that represents a top-level applicacation
+ */
+export function registerApplication(name: string, Component: React.FunctionComponent<any>) {
+  // TODO - does it really make sense that every one of these gets the same redux store? Might we make that configurable?
+  //        or maybe it belongs up a level in the app component being passed in. One argument for here, is if we ever want
+  //        to share a single redux store across multiple embedded apps. One argument for up a level, is not every embedded
+  //        app might need it, and not every one might need the same reducers.
```

Comment:
@darrensiegel An unresolved question came up during our internal review on what to do with this and wanted to know if you had input. No rush, can wait until you're back.

Reply:
At the moment (and when this was all initially built) every "app" needed the same common Redux store.  This was an easy way to allow some common reusable components such as the Media Library to work, out of the box, with any app.  Until some new app has a need for its own custom store, I would not change any of this.